stor/dx.py

```diff
@@ -824,12 +828,60 @@ def upload(self, to_upload, **kwargs):
                 raise NotFoundError('Source path ({}) does not exist. Please provide '
                                     'a valid source'.format(upload_obj.source))
 
-    def open(self, mode='r', encoding=None):
+    def open(self, mode='r', encoding=None, **kwargs):
         """
-        Opens a OBSFile that can be read or written to and is uploaded to
-        the remote service.
+        Args:
+            mode (str): The mode of the resource. Can be "r" for
+                reading the resource, "w" for writing the
+                resource, and "a" for appending to the open resource.
+            encoding (str): the text encoding to use on read/write, defaults to
+                ``locale.getpreferredencoding(False)`` if not set. We *strongly* encourage you to
```

Comment:
why add this sentence? would you generally expect encodings that are not utf-8 or ascii? (i.e., something compatible with utf-8).

Additionally, this only applies when writing text data, correct?

Reply:
Sorry, this docstring was carried over from OBSFile. I modified it. And yes, the encoding applies only when reading/writing text data in Python 3.